js/pro/binance.js

```diff
@@ -42,6 +42,14 @@ module.exports = class binance extends binanceRest {
                 },
             },
             'options': {
+                'streamLimits': {
+                    'spot': 1024,
+                    'margin': 1024,
+                    'future': 200,
+                    'delivery': 200,
+                },
+                'streamBySubscriptionsHash': {},
```

Comment:
after a connection close/error we have to clean this object otherwise we're not completely resetting the state so this could be problematic:

```Javascript
while(true) {
    try {
        await binance.watchTrades("BTC/USDT")
    } catch (e) {
        // does nothing, will reconnect
    }
}
```
⬆️ each reconnection would use the previous "state"

Check for instance how `hollaex` handles this problem (`onError` and `onClose` override)